ambry-cloud/src/main/java/com/github/ambry/cloud/azure/ConnectionStringBasedStorageClient.java

```diff
@@ -105,11 +107,18 @@ protected BlobServiceClient buildBlobServiceSyncClient(HttpClient httpClient, Co
   @Override
   protected TableServiceClient buildTableServiceClient(HttpClient httpClient, Configuration configuration,
       RetryOptions retryOptions, AzureCloudConfig azureCloudConfig) {
-    return new TableServiceClientBuilder().connectionString(azureCloudConfig.azureTableConnectionString)
-        .httpClient(httpClient)
-        .retryOptions(retryOptions)
-        .configuration(configuration)
-        .buildClient();
+    try {
+      return new TableServiceClientBuilder().connectionString(azureCloudConfig.azureTableConnectionString)
+          .httpClient(httpClient)
+          .retryOptions(retryOptions)
+          .configuration(configuration)
+          .buildClient();
+    } catch (Throwable e) {
+      String err = String.format("Failed to build table service client with connection string %s due to %s",
+          azureCloudConfig.azureTableConnectionString, e.getMessage());
+      logger.error(err);
```

Comment:
logger is actually a much better string formatter than you think. 
You can do 
```
logger.error("Failed to build table service client with connection string {} due to {}",
          azureCloudConfig.azureTableConnectionString, e.getMessage());
```

Reply:
ack